Add :compact formatting option

booty opened the pull request:
I discovered I had a need for `dotiw`... except with more compact output. So I added support for a `:compact` formatting option.

I figured this would be fast and easy to implement. And it was.... until I remembered i18n. =)

From the updated README:

---

## compact

Pressed for space? Or do you just hate all those words and stuff? Try `compact: true`.

    >> distance_of_time_in_words(Time.now, Time.now + 2.year + 1.day + 1.second, compact: true)
    => "2y1d"

Pairs well with `words_connector`, `last_word_connector`, and `two_words_connector` if you can spare just a little more room:

    >> distance_of_time_in_words(Time.now, Time.now + 5.years + 1.day + 23.seconds, words_connector: " ", last_word_connector: " ", two_words_connector: " ", compact: true)
    => "5y 1d 23s"

booty commented:
Any thoughts on this? I realize it's a feature nobody (besides me) asked for.  =)

But I think it could be useful to others...

dblock commented:
This is a little old, but I think it would be a useful extension. @booty, if you're still using this library I'd be happy to review a rebased/fixed up version on top of #105 and merge.

booty commented:
@dblock Thanks for taking a look at this. I've rebased on `master`.

I think the functionality of this PR is pretty useful - there are a lot of situations (dashboards, etc) where I'd much rather see "42s" than "42 seconds."

The actual implementation might need some work.

- The way I integrated things into `methods.rb` is not the prettiest, but I could not think of better way. If this was my project I'd probably be okay with it, as I don't see `methods.rb` really increasing much complexity in the future - but wouldn't blame you if you disagreed. 😀
- I updated the locale files, but as a monolingual English speaker, I was purely guessing. Certainly, some of these are not correct. When in doubt, I removed spaces rather than guessing at an abbreviation.

booty commented on lib/dotiw/locale/ar.yml:
```diff
@@ -38,16 +38,67 @@ ar:
         many: "%{count} عاماً"
         other: "%{count} عام"
       less_than_x: "%{distance} أقل من"
+  dotiw_compact:
```
I am told by a friend-of-a-friend that Arabic does not have "abbreviations" per se.

```
First, let me say my Arabic isn’t good enough to provide an actual answer to your question. A native speaker might totally contradict me and they’d be right. However, Arabic very rarely uses abbreviations like that.

In one sense, the words are already abbreviated, because they lack the short vowels.

The only time I can recall seeing “abbreviations” is in the formation of Arabic acronyms (FATAH, HAMAS, etc) which is its own art and not really the same thing.

So: no real answer, but maybe some reassurance that there may not be not a commonly recognized equivalent.

Example: here’s an Arabic calendar. Where we would normally put SMTWTF, they had to squeeze in the entire name of each day.
```

dblock commented:
I'd like to merge this. Care to add a line to https://github.com/radar/distance_of_time_in_words/blob/master/CHANGELOG.md please?

dblock commented on lib/dotiw/action_view/helpers/date_helper.rb:
```diff
@@ -15,6 +15,7 @@ def distance_of_time_in_words(from_time, to_time = 0, include_seconds_or_options
 
       def distance_of_time_in_percent(from_time, current_time, to_time, options = {})
         options[:precision] ||= 0
+        options = options_with_scope(options)
```
We should fix the fact that we're modifying incoming `options` hash here and in other places. For a separate PR if you feel motivated, https://github.com/radar/distance_of_time_in_words/issues/113.

dblock commented:
Looks like I don't have rights to add contributors after all, @radar I suggest adding @booty here and rubygems (he did email me to confirm he wants to help and seems to know what he's doing :).

radar commented:
Can do, just need @booty's RubyGems email. Is it the same as the one on your profile? `johnedmundrose@gmail.com`?

dblock commented:
> Can do, just need @booty's RubyGems email. Is it the same as the one on your profile? `johnedmundrose@gmail.com`?

Looks like it.
